pkg/obfuscate/sql_tokenizer.go

```diff
@@ -473,10 +473,15 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 			// modulo operator (e.g. 'id % 8')
 			return TokenKind(ch), tkn.bytes()
 		case '$':
-			if isDigit(tkn.lastChar) {
-				// TODO(gbbr): the first digit after $ does not necessarily guarantee
-				// that this isn't a dollar-quoted string constant. We might eventually
-				// want to cover for this use-case too (e.g. $1$some text$1$).
+			if isDigit(tkn.lastChar) || tkn.lastChar == '?' {
+				// TODO(knusbaum): Valid dollar quote tags start with alpha characters and contain no symbols.
+				// For some reason I cannot find this in the postgres documentation.
```

Comment:
Is this the documentation you are looking for? https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING